### PR TITLE
Records primer

### PR DESCRIPTION
**Pull request by mppf**
I noticed that we don't have a records primer, so this PR adds one.

Reviewed by @benharsh - thanks!

### CODE REVIEW

File: test/release/examples/primers/records.chpl [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+//
+// This primer covers the declaration and uses of records.
+//
+
+/*
+  
+   A record is a type that can contain variables and constants,
+   called fields, as well as functions and iterators called methods.
+   Records have many similaties to classes, but there are several important
```
Comment (benharsh) [inline test/release/examples/primers/records.chpl]: "similarities"

File: test/release/examples/primers/records.chpl [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+// The record ``Color`` doesn't declare any initializers, and so the
+// compiler-generated one is used. The compiler-generated initializer
+// has an argument for each field in the record.
+
+// The result of ``new`` can be captured in a variable:
+var taupe = new Color(179, 139, 109);
+writeln(taupe); // output: (red = 179, green = 139, blue = 109)
+
+// Alternatively, a variable can be declared using the record type
+// without initializing it.
```
Comment (benharsh) [inline test/release/examples/primers/records.chpl]: Maybe say 'without explicitly initializing it'?

File: test/release/examples/primers/records.chpl [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+// Note that inside of a method, the fields may be accessed
+// by name. Additionally `this` refers to the entire record.
+proc Color.luminance() {
+  return 0.2126*red + 0.7152*green + 0.0722*blue;
+}
+writeln(taupe.luminance());
+
+// Here is an example of a record to store a point where
+// a method is declared inside of the record declaration.
+// This record also demonstrates a a few initializers.
```
Comment (benharsh) [inline test/release/examples/primers/records.chpl]: "a a"

File: test/release/examples/primers/records.chpl [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    this.y = from.y;
+    this.z = from.z;
+  }
+  proc magnitude() {
+    return sqrt(x*x + y*y + z*z);
+  }
+}
+
+// Let's create a default-initialized Point
+var emptyPoint:Point; // outputs Point default-init
```
Comment (benharsh) [inline test/release/examples/primers/records.chpl]: Maybe use a colon like the other comments (e.g. "outputs: "), or wrap the output in quotes?

File: test/release/examples/primers/classes.chpl [hunk 5]
```diff
@@ -17,10 +17,11 @@ class C {
...
 
 //
-// Class variables can be declared using the ``new`` keyword to call the
-// constructor for the class ``C``.  The default constructor has an argument for
-// each field in the class.  Once a class has been constructed, its methods
-// can be called.
+// The ``new`` keyword creates an instance of a class by calling an
+// initializer. The class ``C`` above does not declare any initializers,
+// and so the compiler-generated one is used. The compiler-generated
+// initializer has an argument for each field in the class. Once a class
```
Comment (cassella) [inline test/release/examples/primers/classes.chpl]: Do you want to say anything about the intializer's arguments having default values?

Comment (mppf) [reply test/release/examples/primers/classes.chpl]: Not in this one. I'm hoping that in the future we have an initializers primer where such a thing would be covered.

**Pull request merged**